lib/api/readable.js

```diff
@@ -146,15 +146,39 @@ module.exports = class BodyReadable extends Readable {
 
   async dump (opts) {
     let limit = opts && Number.isFinite(opts.limit) ? opts.limit : 262144
+    const signal = opts && opts.signal
+    function throwIfAborted () {
+      if (!signal) { return }
+      if (signal.throwIfAborted) {
+        signal.throwIfAborted()
+      } else {
+        if (signal.aborted) {
+          // DOMException not available < v17.0.0
+          const err = new Error('The operation was aborted')
+          err.name = 'AbortError'
+          throw err
+        }
+      }
+    }
+    if (signal) {
+      if (typeof signal !== 'object' || !('aborted' in signal)) {
+        throw new InvalidArgumentError('signal must be an AbortSignal')
+      }
+      throwIfAborted()
+      signal.addEventListener('abort', () => {
+        this.destroy()
+      })
```

Comment:
This event listener needs to be cleaned up or we have a memory leak.

Reply:
Made this `{ once: true }`